Save File On to Device And File Content Test

githubRaj opened the pull request:
New Feature Added that allows user to either export a file to their device or to their email. The ReadFromFileSent test checks if the data is properly exported through a file, by comparing the contents of the file with the data in the widget row before the export. Addresses Issue: #75, #39 .

githubRaj committed:
Tesing Read from exported file

Testing if the content of the file being exported

githubRaj committed:
New Feature For Export

The user now has the option to either save the file on their device or send the file to their email. Addresses Issue #75

githubRaj committed:
Code Refactor

Refactored code for better variable names and maintainability.

githubRaj committed:
Refactored

Additional Readability.

githubRaj committed:
Code Climate Issue Fix

Fix For Code Climate Issue

githubRaj committed:
Merge branch 'master' into ExportAdditionalTest

GurkomalSRao commented on src/main/java/nitezh/ministock/SaveFile.java:
```diff
@@ -0,0 +1,56 @@
+
+        try {
+            final boolean newFile = fileSavedOnDevice.createNewFile();
+            FileOutputStream fOut = new FileOutputStream(fileSavedOnDevice);
```
variable newFile is never used

GurkomalSRao commented on src/main/java/nitezh/ministock/activities/PreferencesActivity.java:
```diff
@@ -69,6 +69,10 @@ of this software and associated documentation files (the "Software"), to deal
 import nitezh.ministock.R;
+
+import nitezh.ministock.SaveFile;
+import nitezh.ministock.Storage;
```
`import nitezh.ministock.Storage;` never used

GurkomalSRao commented on src/test/java/nitezh/ministock/activities/EmailTest.java:
```diff
@@ -1,9 +1,24 @@
 
+import android.os.Environment;
+import android.util.Log;
+
```
in email test you have password hard coded. should be changed

GurkomalSRao committed:
Refactoring #39 #75

Rajwinder was not available to make requested changes. Was not able to
make changes before deadline.

GurkomalSRao committed:
Refactoring Code #39 #75

refactored code. removed hardcoded emails, passwords, apikeys.
made it more secure using buildConfig

GurkomalSRao committed:
Refactoring #39 #75
